api/lib/infrastructure/repositories/certification-assessment-repository.js

```diff
@@ -57,8 +58,9 @@ module.exports = {
     });
   },
 
-  async getByCertificationCourseId(certificationCourseId) {
-    const certificationAssessmentRows = await knex('assessments')
+  async getByCertificationCourseId({ certificationCourseId, domainTransaction = DomainTransaction.emptyTransaction() }) {
+    const knexCon = domainTransaction.knexTransaction || knex;
+    const certificationAssessmentRows = await knexCon('assessments')
```

Comment:
Si on destructure, plus besoin de se ballader le [0]

```suggestion
    const [firstCertificationAssessmentRow] = await knexCon('assessments')
```

puis ensuite 
```
if (firstCertificationAssessmentRow) {

```

Reply:
maintenant que je regarde attentivement ça mérite une petite réécriture